packages/cli/src/tools/installPods.ts

```diff
@@ -43,56 +49,83 @@ function runSudo(command: string): Promise<void> {
   });
 }
 
-async function installCocoaPods(loader: ora.Ora) {
-  loader.stop();
-
-  const withGem = 'Yes, with gem (may require sudo)';
-  const withHomebrew = 'Yes, with Homebrew';
+async function promptCocoaPodsInstallationQuestion(): Promise<
+  PromptCocoaPodsInstallation
+> {
+  const promptQuestion = `CocoaPods ${chalk.dim.underline(
+    '(https://cocoapods.org/)',
+  )} ${chalk.reset.bold(
+    'is not installed. CocoaPods is necessary for the iOS project to run correctly. Do you want to install it?',
+  )}`;
+  const installWithGem = 'Yes, with gem (may require sudo)';
+  const installWithHomebrew = 'Yes, with Homebrew';
 
   const {shouldInstallCocoaPods} = await inquirer.prompt([
     {
       type: 'list',
       name: 'shouldInstallCocoaPods',
-      message: `CocoaPods ${chalk.dim.underline(
-        '(https://cocoapods.org/)',
-      )} ${chalk.reset.bold(
-        'is not installed. CocoaPods is necessary for the iOS project to run correctly. Do you want to install it?',
-      )}`,
-      choices: [withGem, withHomebrew],
+      message: promptQuestion,
+      choices: [installWithGem, installWithHomebrew],
     },
   ]);
 
-  switch (shouldInstallCocoaPods) {
-    case withGem:
-      const options = ['install', 'cocoapods', '--no-document'];
-      loader.start('Installing CocoaPods');
-      try {
-        // First attempt to install `cocoapods`
-        await execa('gem', options);
-      } catch (_error) {
-        // If that doesn't work then try with sudo
-        try {
-          await runSudo(`gem ${options.join(' ')}`);
-        } catch (error) {
-          loader.fail();
-          logger.error(error.stderr);
-
-          throw new Error(
-            `An error occured while trying to install CocoaPods, which is required by this template.\nPlease try again manually: sudo gem install cocoapods.\nCocoaPods documentation: ${chalk.dim.underline(
-              'https://cocoapods.org/',
-            )}`,
-          );
-        }
-      }
-      loader.succeed();
-      break;
-    case withHomebrew:
-      await brewInstall({
-        pkg: 'cocoapods',
-        label: 'Installing CocoaPods',
-        loader,
-      });
-      break;
+  const shouldInstallWithGem = shouldInstallCocoaPods === installWithGem;
+
+  return {
+    installMethod: shouldInstallWithGem ? 'gem' : 'homebrew',
+    // This is used for removing the message in `doctor` after it's answered
+    promptQuestion: `? ${stripAnsi(promptQuestion)} ${
+      shouldInstallWithGem ? installWithGem : installWithHomebrew
+    }`,
+  };
+}
+
+async function installCocoaPodsWithGem() {
+  const options = ['install', 'cocoapods', '--no-document'];
+
+  try {
+    // First attempt to install `cocoapods`
+    await execa('gem', options);
+  } catch (_error) {
+    // If that doesn't work then try with sudo
+    try {
+      await runSudo(`gem ${options.join(' ')}`);
+    } catch (error) {
+      throw error;
```

Comment:
if we re-throw without any side effects around, then there's no need to try/catch